src/v3/src/components/List/ListItemText.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Box } from '@okta/odyssey-react-mui';
+import { FunctionComponent, h } from 'preact';
+
+import { useHtmlContentParser } from '../../hooks';
+
+type ListItemTextProps = {
+  text: string;
+};
+
+const ListItemText: FunctionComponent<ListItemTextProps> = ({ text }) => {
```

Comment:
Not a blocker at all, but is there much value in this component compared to inserting the `useHtmlContentParser` hook in `List`? This is essentially a thin wrapper around that

Reply:
was not able to use the useHtmlContentParser hook in List without getting the 
`React Hook "useHtmlContentParser" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks` error

Reply:
In `List.tsx` there already is usage of a hook `useOdysseyDesignTokens()` - why is this hook not allowed?

Reply:
looks like because we use `useHtmlContentParser` within a callback for the array map on line 100.  Which in that case the hooks call order cannot be guaranteed: https://www.dhiwise.com/post/fixing-the-react-hook-cannot-be-called-inside-a-callback#:~:text=This%20is%20because%20React%20relies%20on%20the%20order%20of%20hook%20calls%20to%20track%20the%20state%20and%20effects%20associated%20with%20each%20hook%20properly.%20When%20you%20place%20a%20hook%20inside%20a%20callback%20function%2C%20the%20hook%27s%20call%20order%20can%20no%20longer%20be%20guaranteed%2C%20as%20the%20callback%20function%20may%20not%20be%20called%20on%20every%20render.

Reply:
im questioning whether `useHtmlContentParser` even needs to be a hook, seems to just act like a normal function that parses some text on the spot.  Maybe we could  just treat it as a regular function?

Reply:
@leemchale-okta you're right - let's move this to another directory (maybe there's something for utils) and rename it so it doesn't get picked up by the linter for hooks rules.

Reply:
yeah agreed sounds good will get that refactored.

slack discussion ref: https://okta.slack.com/archives/C04NYNLLRDL/p1721768477843139